File: protect-secrets/ssm.py
import boto3


class SSMInterface:

    def __init__(self):
        self.client = boto3.client('ssm')

    def upload(self, **kwargs):
        response = self.client.put_parameter(
            Name=kwargs['name'],
            Value=kwargs['value'],
            Type='SecureString',
            Overwrite=kwargs['overwrite'],
        )
        return response

    def download(self, **kwargs):
        response = self.client.get_parameter(
            Name=kwargs['name'],
            WithDecryption=True
        )
        return response

'''
    # example usage:

    ssm = SSMInterface()
    ssm.upload(name='/some/key', value='test', overwrite=True)
    ssm.download(name='/some/key', decryption=True)
'''

File: protect-secrets/ssm.js
const AWS = require('aws-sdk');
const ssm = new AWS.SSM();

const upload = async (param) => {
    await ssm.putParameter(param).promise();
    return Promise.resolve(true);
};

const download = async (paramName) => {
    try {
        const result = await ssm.getParameter({Name: paramName,WithDecryption: true}).promise();
        if (result) {
            return result.Parameter.Value;
        }
    } catch (error) {
        return error;
    }
};

module.exports = {upload, download};
